crates/chain-config/src/config/state.rs

```diff
@@ -52,61 +49,85 @@ pub const TESTNET_WALLET_SECRETS: [&str; 5] = [
 
 pub const STATE_CONFIG_FILENAME: &str = "state_config.json";
 
-// TODO: do streaming deserialization to handle large state configs
-#[serde_as]
-#[skip_serializing_none]
 #[derive(Clone, Debug, Default, Deserialize, Serialize, Eq, PartialEq)]
 pub struct StateConfig {
     /// Spendable coins
-    pub coins: Option<Vec<CoinConfig>>,
-    /// Contract state
-    pub contracts: Option<Vec<ContractConfig>>,
+    pub coins: Vec<CoinConfig>,
```

Comment:
I'm a little confused with the nominclature. It looks like its a holdover from before, but I want to make sure I understand. Some of these fields are `Vec<SomeConfig>`, but then the docstring says "state".

To me config and state are different things--apps with the same configs could hold different state. If `Config` is the right term here, I'm confused what it would mean to have a vec of configs in general.

I think the names /docstrings/both need some clarity here.

Reply:
My understanding was these are parts of the initial state config, so if you were to use the same configs you would end up with the same initial state. 
I'm not sure what naming convention to go with to best clarify this, for now I just removed "state" from the docstring. If you have some ideas how to improve clarity I'm open for it.

Reply:
Sounds good enough for now. I don't want to slow you down too much.